From 6c967cebd213d1820cd1af428db5449b1af9aec1 Mon Sep 17 00:00:00 2001
From: Hugh Waite <hugh.waite@acision.com>
Date: Tue, 12 Aug 2014 19:52:27 +0100
Subject: [PATCH] outbound: Fix memory leak

- Free parameters parsed when checking for ob markers
(cherry picked from commit 392689597376eb6f047c12504bcce367f5940260)
---
 modules/outbound/ob_mod.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules/outbound/ob_mod.c b/modules/outbound/ob_mod.c
index 038c5fa..aae1e10 100644
--- a/modules/outbound/ob_mod.c
+++ b/modules/outbound/ob_mod.c
@@ -385,6 +385,8 @@ static int use_outbound_non_reg(struct sip_msg *msg)
 			LM_ERR("parsing Route-URI parameters\n");
 			return 0;
 		}
+		/* Not interested in param body - just the hooks */
+		free_params(params);
 
 		if (hooks.uri.ob)
 		{
@@ -450,6 +452,9 @@ static int use_outbound_non_reg(struct sip_msg *msg)
 			LM_ERR("parsing Contact-URI parameters\n");
 			return 0;
 		}
+		/* Not interested in param body - just the hooks */
+		free_params(params);
+
 		if (hooks.contact.ob)
 		{
 			LM_DBG("found ;ob parameter on Contact-URI - outbound"
-- 
2.1.0.rc1

